### PR TITLE
Harden InterproceduralAnalysisConfiguration to prevent null tree argu…

### LINKED ISSUE #4491: ArgumentNullException in SourceTriggeredTaintedDataAnalyzerBase for VB assembly attributes
## Unit test for ReviewCodeForSqlInjectionVulnerabilitiesTests
```csharp
        [Fact]
        public async Task AssemblyAttributeRegressionTest()
        {
            await VerifyVisualBasicWithDependenciesAsync(@"<Assembly: System.Reflection.AssemblyTitle(""Title"")>");
        }
```

## Exception call stack

```
    // error AD0001: Analyzer 'Microsoft.NetCore.Analyzers.Security.ReviewCodeForSqlInjectionVulnerabilities' threw an exception of type 'System.ArgumentNullException' with message 'Value cannot be null. (Parameter 'key')'.
    Exception occurred with following context:
    Compilation: TestProject
    ISymbol:  (Namespace)
    
    System.ArgumentNullException: Value cannot be null. (Parameter 'key')
       at System.Collections.Immutable.Requires.FailArgumentNullException(String parameterName)
       at System.Collections.Immutable.ImmutableDictionary`2.TryGetValue(TKey key, TValue& value)
       at Analyzer.Utilities.AggregateCategorizedAnalyzerConfigOptions.TryGetOptionValue[T](String optionName, OptionKind kind, SyntaxTree tree, DiagnosticDescriptor rule, TryParseValue`1 tryParseValue, T defaultValue, T& value) in c:\roslyn-analyzers\src\Utilities\Compiler\Options\AggregateCategorizedAnalyzerConfigOptions.cs:line 101
       at Analyzer.Utilities.AggregateCategorizedAnalyzerConfigOptions.GetOptionValue[T](String optionName, SyntaxTree tree, DiagnosticDescriptor rule, TryParseValue`1 tryParseValue, T defaultValue, OptionKind kind) in c:\roslyn-analyzers\src\Utilities\Compiler\Options\AggregateCategorizedAnalyzerConfigOptions.cs:line 79
       at Analyzer.Utilities.AnalyzerOptionsExtensions.GetNonFlagsEnumOptionValue[TEnum](AnalyzerOptions options, String optionName, DiagnosticDescriptor rule, SyntaxTree tree, Compilation compilation, TEnum defaultValue, CancellationToken cancellationToken) in c:\roslyn-analyzers\src\Utilities\Compiler\Options\AnalyzerOptionsExtensions.cs:line 187
       at Analyzer.Utilities.AnalyzerOptionsExtensions.GetInterproceduralAnalysisKindOption(AnalyzerOptions options, DiagnosticDescriptor rule, SyntaxTree tree, Compilation compilation, InterproceduralAnalysisKind defaultValue, CancellationToken cancellationToken) in c:\roslyn-analyzers\src\Utilities\FlowAnalysis\Options\AnalyzerOptionsExtensions_FlowAnalysis.cs:line 31
       at Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.InterproceduralAnalysisConfiguration.Create(AnalyzerOptions analyzerOptions, DiagnosticDescriptor rule, SyntaxTree tree, Compilation compilation, InterproceduralAnalysisKind defaultInterproceduralAnalysisKind, CancellationToken cancellationToken, UInt32 defaultMaxInterproceduralMethodCallChain, UInt32 defaultMaxInterproceduralLambdaOrLocalFunctionCallChain) in c:\roslyn-analyzers\src\Utilities\FlowAnalysis\FlowAnalysis\Framework\DataFlow\InterproceduralAnalysisConfiguration.cs:line 62
       at Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.InterproceduralAnalysisConfiguration.Create(AnalyzerOptions analyzerOptions, ImmutableArray`1 rules, SyntaxTree tree, Compilation compilation, InterproceduralAnalysisKind defaultInterproceduralAnalysisKind, CancellationToken cancellationToken, UInt32 defaultMaxInterproceduralMethodCallChain, UInt32 defaultMaxInterproceduralLambdaOrLocalFunctionCallChain) in c:\roslyn-analyzers\src\Utilities\FlowAnalysis\FlowAnalysis\Framework\DataFlow\InterproceduralAnalysisConfiguration.cs:line 111
       at Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.InterproceduralAnalysisConfiguration.Create(AnalyzerOptions analyzerOptions, ImmutableArray`1 rules, ISymbol symbol, Compilation compilation, InterproceduralAnalysisKind defaultInterproceduralAnalysisKind, CancellationToken cancellationToken, UInt32 defaultMaxInterproceduralMethodCallChain, UInt32 defaultMaxInterproceduralLambdaOrLocalFunctionCallChain) in c:\roslyn-analyzers\src\Utilities\FlowAnalysis\FlowAnalysis\Framework\DataFlow\InterproceduralAnalysisConfiguration.cs:line 93
       at Microsoft.NetCore.Analyzers.Security.SourceTriggeredTaintedDataAnalyzerBase.<>c__DisplayClass6_0.<Initialize>b__1(OperationBlockStartAnalysisContext operationBlockStartContext) in c:\roslyn-analyzers\src\NetAnalyzers\Core\Microsoft.NetCore.Analyzers\Security\SourceTriggeredTaintedDataAnalyzerBase.cs:line 80
       at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__62`6.<ExecuteBlockActionsCore>b__62_2(ValueTuple`5 data)
       at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock[TArg](DiagnosticAnalyzer analyzer, Action`1 analyze, TArg argument, Nullable`1 info)
    -----
```

### PR DESCRIPTION
**Pull request by mavasani**
…ment

Change the API argument for `InterproceduralAnalysisConfiguration.Create` to avoid null syntax tree being passed into it.

Fixes #4491. ~~I will create a separate PR to port the fix to 2.9.x branch~~

### CODE REVIEW

File: src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/InterproceduralAnalysisConfiguration.cs [hunk 1]
```diff
@@ -40,16 +40,16 @@ public struct InterproceduralAnalysisConfiguration : IEquatable<InterproceduralA
         public static InterproceduralAnalysisConfiguration Create(
             AnalyzerOptions analyzerOptions,
             DiagnosticDescriptor rule,
-            ISymbol symbol,
+            ControlFlowGraph cfg,
```
Comment (mavasani) [inline src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/InterproceduralAnalysisConfiguration.cs:43]: This is the core change, everything else in the PR is just boiler plate changes.

**Comment by mavasani** [comment]
> I will create a separate PR to port the fix to 2.9.x branch

@dotpaul - I figured out that this issue does not repro on 2.9.x branch as the support for passing in `symbol` or `tree` into the constructor of `InterproceduralAnalysisConfiguration ` was added post 3.3.x when editorconfig options support was added. Constructor on 2.9.x: https://github.com/dotnet/roslyn-analyzers/blob/ed4a12cadf4cffef27ce8a11f8e21952dfdb5730/src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/InterproceduralAnalysisConfiguration.cs#L40-L47

**Review by dotpaul — APPROVED** [review]
:ship:

**Comment by mavasani** [comment]
Thanks you Paul!

**Pull request merged**